techlibs/quicklogic/synth_quicklogic.cc

```diff
@@ -108,51 +109,94 @@ struct SynthQuickLogicPass : public ScriptPass
         }
 
         if (check_label("coarse")) {
+            run("opt_expr");
+            run("opt_clean");
+            run("check");
+            run("opt");
             run("wreduce -keepdc");
             run("peepopt");
             run("pmuxtree");
-            run("opt");
-            run("memory -nomap");
             run("opt_clean");
+            run("share");
+            run("techmap -map +/cmp2lut.v -D LUT_WIDTH=4");
+            run("opt_expr");
+            run("opt_clean");
+
             run("alumacc");
+            run("opt");
+            run("fsm");
             run("opt -fast");
+            run("memory -nomap");
             run("opt_clean");
-            run("opt -full");
-            run("check");
         }
 
-        if (check_label("map")) {
-            if (check_label("map_bram", "(skip if -nobram)"))
-            {
-               run("memory_bram -rules +/quicklogic/" + family + "_brams.txt");
-               run("techmap -map +/quicklogic/" + family + "_brams_map.v");
-            }
+        if (check_label("map_bram", "(skip if -nobram)")) {
```

Comment:
This is ok but `pp3_brams.txt' and `pp3_brams_map.v` require rework. Addres and data widths defined there do not correspond with capabilities of PP3E RAM cells thus are not optimal.

Reply:
Sure, I took ram related code as a place holder as Tarachand is currently working on that. So, this part can be changed as you or Tarachand suggest.